ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/host/Webserver.java

```diff
@@ -23,39 +23,42 @@ public class Webserver {
 
   private static final Pattern HTTP_GET_PATTERN = Pattern.compile("GET /(?<resource>.*) HTTP");
   private static final ILog LOGGER = Platform.getLog(Webserver.class);
+  private static final Resource NULL_RESOURCE = new Resource(null, null, null);
 
   private Map<String, Resource> resources;
   private ExecutorService executor;
   private ServerSocket serverSocket;
-  private CompletableFuture<Void> whenStarted = new CompletableFuture<>();
 
   public Webserver(Map<String, Resource> resources) {
     this.resources = resources;
   }
 
   public Webserver start() {
     serverSocket = createServerSocket();
-    executor = Executors.newSingleThreadExecutor();
+    executor = Executors.newFixedThreadPool(2);

```

Comment:
Question: We now are passing a backlog size when creating the `ServerSocket`. Do we still need multiple threads? The reason we changed this was that we had lost requests and as a result the application failed to start.

Reply:
Proposal: Replace this custom web server with a server provided by Eclipse in a separate commit.

Reply:
Does Eclipse provide a webserver?

Reply:
> Question: We now are passing a backlog size when creating the `ServerSocket`. Do we still need multiple threads? The reason we changed this was that we had lost requests and as a result the application failed to start.

The custom backlog size didn't solve the issue, so I would keep the multiple threads for now, and later on we will see if we switch anyway to the webserver metioned by @marchof anyway, since that one I assume it is more stable and hopefully  address already these kind of problems.

Reply:
> Question: We now are passing a backlog size when creating the `ServerSocket`. Do we still need multiple threads? The reason we changed this was that we had lost requests and as a result the application failed to start.

The custom backlog size didn't solve the issue, so I would keep the multiple threads for now, and later on we will see if we switch anyway to the webserver provided by @marchof anyway, since that one I assume it is more stable and hopefully  address already these kind of problems.

Reply:
For example org.eclipse.help.ui uses a web server. I will check whether there is a default solution to this in the Eclipse platform.